naming-scheme.rst

```diff
@@ -0,0 +1,82 @@
+New Python package naming scheme
+================================
+
+.. include:: snippets/naming_scheme.inc
+
+.. note::
+	This section is related to the renaming of Python **binary RPM** packages to avoid using the ``python-`` prefix without a version. Changing the package/SRPM name is not required.
+
+Why is this important?
+----------------------
+
+When the time comes and ``python`` means Python 3 in Fedora, installing a ``python-<srcname>`` package will imply a Python 3 version of this package. This is planned for 2020, when upstream support for Python 2 ends. To achieve this, all Python binary RPM packages have to follow the new naming scheme and use the ``%python_provide`` macro, devised to make the switch easier. However, we are still far away from achieving this goal.
+
+Using the outdated naming scheme in your subpackage names or run-time/build-time requirements might cause a range of issues when the switch happens.
+
+What needs to be changed?
+-------------------------
+
+Check the names of binary RPM packages you are building from your SRPM, and if you use one of the following naming schemes, you'll find instructions on how to fix it in the `section below <required_changes_>`_.
+
+Common naming scheme violations
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+See the following naming schemes which violate the current naming guidelines:
+
+.. list-table:: 
+   :header-rows: 1
+   :stub-columns: 1
+
+   *  -  SRPM
+      -  Binary RPMs built
+      -  Violaion
+   *  -  python-<srcname>
+      -  | python-<srcname>
+      -  unversioned `python-` prefix in the binary package
+   *  -  python-<srcname>
+      -  | python-<srcname>
+         | python3-<srcname>
+      -  unversioned `python-` prefix in the binary package
+   *  -  <srcname>
+      -  | <srcname>
+         | python3-<srcname>
+      -  missing `python2-` prefix in the binary package
+
+.. _required_changes:
+
+Required changes
+^^^^^^^^^^^^^^^^
+
+Add a ``%package`` section for the Python 2 subpackage
+""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+To rename the binary RPM package to ``python2-<srcname>``, you should build it as a subpackage with an appropriate name. Make sure to move all related runtime requirements from the main package to the new subpackage and use the `%python_provide`_ macro, which will provide both ``python-<srcname>`` and ``python2-<srcname>`` until the switch to Python 3 happens.
+
+The change should look like this:
+
+.. literalinclude:: specs/module.spec
+	:diff: specs/module.spec.orig
+	:lines: 15-19,21-33
```

Comment:
Don't know how this macro works, but it only lists the description (content, not `%description`) being added twice, never removed, and it feels wrong.